### ISSUE TITLE
Support for HTTP API tagging

### ISSUE THREAD

**Issue by farski** [awslabs/serverless-application-model#1478]
**Description:**

I don't believe there is currently any way to add resource tags to an `AWS::Serverless::HttpApi` resource[1]. The [SAM specification](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-httpapi.html) does not include anything, and I don't think it's possible to do it through an OpenAPI definition (though admittedly my experience with OpenAPI/Swagger is limited. I have read through the docs, specifically the `info` section, and did not find anything).

[1] It turns out you can add a `tags` list to the OpenApi `BodyDefinition` and those tags will get applied to the HttpApi, but they don't get applied to default stage, and there may be better ways of supporting tags, as discussed below.

**Comment by farski** [comment]
Digging into the code on this a bit (which is probably fraught because I've never looked at the SAM codebase before), it does appear as though there is some level of support for tags on `AWS::Serverless::HttpApi`resources [here](https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/sam_resources.py#L871). The `to_cloudformation` function passes tags into the [`HttpApiGenerator`](https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/sam_resources.py#L902).

Eventually the tags make their way to the [`open_api`](https://github.com/awslabs/serverless-application-model/blob/c5221fcf732475bcc59c5238f963dc95c0033729/samtranslator/open_api/open_api.py#L375) class and get added to the OpenApi definition as `_X_APIGW_TAG_VALUE` extensions.

In fact, the **only** way to get tags to work with an HTTP API is if the resource definition include an inline [`DefinitionBody`](https://github.com/awslabs/serverless-application-model/blob/c5221fcf732475bcc59c5238f963dc95c0033729/samtranslator/model/api/http_api_generator.py#L205).

It doesn't seem like it's necessary to handling tagging this way, since the underlying [`AWS::ApiGatewayV2::Api`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-api.html#cfn-apigatewayv2-api-tags) resource natively supports tags, as does [`AWS::ApiGatewayV2::Stage`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-stage.html#cfn-apigatewayv2-stage-tags).

So perhaps there is support for this already, but it's just not documented.

All that being said, when I try to deploy this SAM template…

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Resources:
  HttpApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      Tags:
        TagName2: TagValue2
```

…it fails with the error `Property validation failure: [Value of property {/Tags} does not match type {Map}]`, even though that should be treated as a `Map`. Alternatively, this CloudFormation template…

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  HttpApi:
    Type: AWS::ApiGatewayV2::Api
    Properties:
      Name: TEST TEST
      Description: TEST TEST
      ProtocolType: HTTP
      Target: "https://www.example.com/"
      Tags:
        TagName1: TagValue1
```

…does work as expected. So I'm not quite sure where the disconnect is.

**Comment by adrai** [comment]
fyi: at least in DefinitionBody it seems to work:
![image](https://user-images.githubusercontent.com/1086194/75665558-a1e01380-5c74-11ea-9512-e44af9405b63.png)

**Comment by farski** [comment]
@adrai Yes that does work, but as I mentioned it doesn't trickle down to the `$default` stage, which make that more or less untaggable.

**Comment by keetonian** [comment]
@farski we're working on Http Api features for this next release, and adding tags to this resource (utilizing OpenApi) is a new feature. I like the suggestion passing through these tags to the `Stage` resource.

**Comment by ShreyaGangishetty** [comment]
This is released in [v1.22.0](https://github.com/awslabs/serverless-application-model/releases/tag/v1.22.0)
